hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpSync.java

```diff
@@ -59,6 +60,9 @@
   //
   private EnumMap<SnapshotDiffReport.DiffType, List<DiffInfo>> diffMap;
   private DiffInfo[] renameDiffs;
+  // entries which are marked deleted because of rename to a excluded target
+  // path
+  private DiffInfo[] deletedByExclusionDiffs;
```

Comment:
Let's use List<DiffInfo>.  Then we don't have to convert to an array in the code below.